File: Task2/task2.cpp
#include <iostream>
#include <vector>
#include <string>
#include <set>
#include <sstream>

using namespace std;

const string morseCodes[26] = {
    ".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---",
    "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-",
    "..-", "...-", ".--", "-..-", "-.--", "--.."
};

int main() {
    cout << "Введите слова:\n";

    string line;
    getline(cin, line);

    istringstream iss(line);
    string word;

    set<pair<int, int>> unique; // (точки, тире)

    while (iss >> word) {

        int dots = 0;
        int dashes = 0;

        for (char c : word) {
            c = tolower(c);
            string code = morseCodes[c - 'a'];

            for (char s : code) {
                if (s == '.') dots++;
                else if (s == '-') dashes++;
            }
        }

        unique.insert({dots, dashes});
    }

    cout << "Уникальных слов: " << unique.size() << endl;

    return 0;
}

File: Task2/task2.rs
use std::collections::HashSet;
use std::io;

fn main() {
    let morse: [&str; 26] = [
        ".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---",
        "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-",
        "..-", "...-", ".--", "-..-", "-.--", "--.."
    ];

    println!("Введите слова:");
    let mut line = String::new();
    io::stdin().read_line(&mut line).unwrap();

    let mut unique = HashSet::new();

    for word in line.split_whitespace() {
        let mut dots = 0;
        let mut dashes = 0;

        for ch in word.chars() {
            if ch.is_ascii_alphabetic() {
                let idx = (ch.to_ascii_lowercase() as u8 - b'a') as usize;
                let code = morse[idx];
                dots += code.chars().filter(|&c| c == '.').count();
                dashes += code.chars().filter(|&c| c == '-').count();
            }
        }

        unique.insert((dots, dashes));
    }

    println!("Уникальных слов: {}", unique.len());
}